metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -87,13 +89,16 @@ public PartitionChangeBuilder(
         Uuid topicId,
         int partitionId,
         IntPredicate isAcceptableLeader,
-        MetadataVersion metadataVersion
+        MetadataVersion metadataVersion,
+        boolean zkMigrationEnabled
```

Comment:
I think this should be descriptive, like "bumpLeaderEpochOnIsrShrink". Then it can be initialized to `!metadataVersion.isSkipLeaderEpochBumpSupported()`. Then we can have a setter function that overrides it, and we can do
```
if (zkMigrationEnabled) {
  builder.setBumpLeaderEpochOnIsrShrink(true);
}
```